examples/Sinusoid/emulate.jl

```diff
@@ -0,0 +1,241 @@
+x_test_unconstrained = Emulators.transform_constrained_to_unconstrained(prior, x_test)
+y_gp, y_gp_covs = Emulators.predict(emulator_gp, x_test_unconstrained, transform_to_real = true);
+
+# This predicts an output and a covariance matrix.
+# Estimate root mean squared error
+rmse_gp = sqrt.(mean((y_gp - y_true) .^ 2, dims = 2))
+println("RMSE GP:", rmse_gp)     # [1.0188362036675438; 0.10148451670974754;;]
+
+# Get standard deviations from covariance matrix
+y_gp_std1 = [sqrt(y_gp_cov[1, 1]) for y_gp_cov in y_gp_covs]
+y_gp_std2 = [sqrt(y_gp_cov[2, 2]) for y_gp_cov in y_gp_covs]
+
+# Plots of y_true vs y_gp to check accuracy of emulator (neaten up)
+p1 = plot(y_true[1, :], y_gp[1, :]; yerr = y_gp_std1, seriestype = :scatter, label = "Gaussian Process")
+Plots.abline!(1, 0; line = :dash, color = :black, label = :false)
+xlabel!("True y1")
+ylabel!("Predicted y1")
+p2 = plot(y_true[2, :], y_gp[2, :]; yerror = y_gp_std2, seriestype = :scatter, label = "Gaussian Process")
+Plots.abline!(1, 0; line = :dash, color = :black, label = :false)
+xlabel!("True y2")
+ylabel!("Predicted y2")
+p = plot(p1, p2, layout = (1, 2))
+savefig(p, joinpath(data_save_directory, "sinusoid_GP_emulator.png"))
+
+# Plot outputs as contours
+# f returns emulator prediction only (no variance) given x1 and x2, for output j=1 or 2
+f(x1, x2; j = 1) = Emulators.predict(
+    emulator_gp,
+    Emulators.transform_constrained_to_unconstrained(prior, hcat(x1, x2)'),
+    transform_to_real = true,
+)[1][
+    j,
+    :,
+][1]
+
+# Create mesh for contour over full ranges of parameter space
+x_range = range.(minimum(inputs, dims = 2), maximum(inputs, dims = 2), length = 50)
+
+# GP predicted contours for output 1, overlaid with model data as scatter points
+z = @. f(x_range[1]', x_range[2]; j = 1)
+p1 = contour(x_range[1], x_range[2], z; fill = true)
+plot!(inputs[1, :], inputs[2, :]; seriestype = :scatter, zcolor = outputs[1, :], label = :false)
+xlabel!("θ1")
+ylabel!("θ2")
+title!("y1")
+# GP predicted contours for output 2, overlaid with model data as scatter points
+z = @. f(x_range[1]', x_range[2]; j = 2)
+p2 = contour(x_range[1], x_range[2], z; fill = true)
+plot!(inputs[1, :], inputs[2, :]; seriestype = :scatter, zcolor = outputs[2, :], label = :false)
+xlabel!("θ1")
+ylabel!("θ2")
+title!("y2")
+# Put subplots together into on figure
+p = plot(p1, p2, layout = (1, 2))
+savefig(p, joinpath(data_save_directory, "sinusoid_GP_emulator_contours.png"))
+
+
+## Random Features
+# An alternative to Gaussian process for the emulator
+# We have two input dimensions and two output dimensions.
+input_dim = 2
+output_dim = 2
+# Select number of features
+n_features = 60
+nugget = 1e-12
+kernel_structure = SeparableKernel(LowRankFactor(2, nugget), OneDimFactor())
+
+# Create random features
+optimizer_options = Dict(
+    "n_ensemble" => 20,
+    "scheduler" => EKP.DataMisfitController(on_terminate = "continue"),
+    "n_iteration" => 50,
```

Comment:
add "verbose" => true, this will print errors so you can easily see what a reasonable number of iterations/ensemble size is